python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -536,3 +563,150 @@ def to_arrow(self):
                 arrow_polygons,
             ],
         )
+
+    def contains(self, other, align=True):
+        """Compute from a set of points and a set of polygons which points fall
+        within each polygon. Note that `polygons_(x,y)` must be specified as
+        closed polygons: the first and last coordinate of each polygon must be
+        the same.
+
+        This implements `.contains_properly`, which shares a large
+        space of correct cases with `GeoPandas.contains` but they do not
+        produce identical results. In the future we will use intersection
+        testing to match .contains behavior.
+
+        Parameters
+        ----------
+        test_points_x
+            x-coordinate of test points
+        test_points_y
+            y-coordinate of test points
+        poly_offsets
+            beginning index of the first ring in each polygon
+        poly_ring_offsets
+            beginning index of the first point in each ring
+        poly_points_x
+            x closed-coordinate of polygon points
+        poly_points_y
+            y closed-coordinate of polygon points
+
+        Examples
+        --------
+
+        Test if a polygon is inside another polygon:
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0.5, 0.5)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        dtype: bool
+
+
+        Test whether 3 points fall within either of two polygons
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0, 0)],
+            [Point(0, 0)],
+            [Point(0, 0)],
+            [Point(-1, 1)],
+            [Point(-1, 1)],
+            [Point(-1, 1)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        1    False
+        2    False
+        3     True
+        4     True
+        5     True
+        dtype: bool
+
+        Note
+        ----
+        input Series x and y will not be index aligned, but computed as
+        sequential arrays.
```

Comment:
Do you mean test_points_x and test_points_y?

Reply:
No, the docs between `core/geoseries.py::contains` and `core/binops/contains.py::contains` were mixed up.

Reply:
Please take a look at the docs between the two functions which should now be fixed.

Reply:
So are x and y members of GeoSeries? It reads like you are referring to input parameter names, but the input is `other`.